xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -725,6 +744,72 @@ ProcColorSpaces CProcessorHD::CalculateDXGIColorSpaces(const DXGIColorSpaceArgs&
   const bool limited = DX::Windowing()->UseLimitedColor();
   const bool supportHDR = DX::Windowing()->IsHDROutput() && (m_bSupportHDR10Limited || !limited);
 
-  return ProcColorSpaces{GetDXGIColorSpaceSource(csArgs, supportHDR, m_bSupportHLG),
-                         GetDXGIColorSpaceTarget(csArgs, supportHDR, limited)};
+  return ProcColorSpaces{
+      GetDXGIColorSpaceSource(csArgs, supportHDR, m_bSupportHLG, m_BT2020TopLeft),
+      GetDXGIColorSpaceTarget(csArgs, supportHDR, limited)};
+}
+
+bool CProcessorHD::IsBT2020Supported()
+{
+  ComPtr<ID3D11VideoDevice> pVideoDevice;
+  ComPtr<ID3D11VideoProcessorEnumerator> pEnumerator;
+  ComPtr<ID3D11VideoProcessorEnumerator1> pEnumerator1;
+  ComPtr<ID3D11Device> pD3DDevice = DX::DeviceResources::Get()->GetD3DDevice();
+
+  if (FAILED(pD3DDevice.As(&pVideoDevice)))
+  {
+    CLog::LogF(LOGDEBUG, "failed to get video device.");
+    return false;
+  }
+
+  D3D11_VIDEO_PROCESSOR_CONTENT_DESC desc = {};
+  desc.InputWidth = 3840;
+  desc.InputHeight = 2160;
+  desc.OutputWidth = 3840;
+  desc.OutputHeight = 2160;
+
+  if (FAILED(pVideoDevice->CreateVideoProcessorEnumerator(&desc, &pEnumerator)))
+  {
+    CLog::LogF(LOGDEBUG, "failed to create Video Enumerator.");
+    return false;
+  }
+
+  if (FAILED(pEnumerator.As(&pEnumerator1)))
+  {
+    CLog::LogF(LOGDEBUG, "failed to create Video Enumerator1.");
```

Comment:
Should this use `LOGERROR` (same with a couple other places)?

Reply:
I put it in debug specifically because it is not supported in Windows 8 and it is normal for it to fail.
In fact we want it to fail because in Windows 8 we can't use `ID3D11VideoContext1` either and fallback code path not supports BT.2020.

Reply:
Is it possible to determine the OS support early so we can return early? Then these log lines aren't so superfluous.

Reply:
It's related to previous Q/A. For this interface it would be possible but IIRC not the recommended approach by MS and the test still has to happen even when you know the OS is Windows 10 or you could end up dereferencing a null pointer.

It's not really an error, just an indication that the code is taking a path or not and has its place at debug log level I think. Maybe a more neutral message like "ID3D11VideoProcessorEnumerator1 is not available" is better?

Reply:
Changed first check to `LOGERROR` and this one to `LOGWARNING` "ID3D11VideoProcessorEnumerator1 is not available. DXVA will not be used."